README.md

```diff
@@ -68,18 +66,126 @@ you are faced with a whole set of new problems:
+inside the `addButtonTapped` method, you can use the [`withDependencies`][withdependencies-docs]
+function to override any dependencies for the scope of one single test. It's as easy as 1-2-3:
+
+```swift
+func testAdd() async throws {
+  let model = withDependencies {
+    // 1️⃣ Override any dependencies that your feature uses.
+    $0.clock = ImmediateClock()
+    $0.date.now = Date(timeIntervalSinceReferenceDate: 1234567890)
+    $0.uuid = .incrementing
+  } operation: {
+    // 2️⃣ Construct the feature's model
+    FeatureModel()
+  }
+
+  // 3️⃣ The model now executes in a controlled environment of dependencies,
+  //    and so we can make assertions against its behavior.
+  try await model.addButtonTapped()
+  XCTAssertEqual(
+    model.items,
+    [
+      Item(
+        id: UUID(uuidString: "00000000-0000-0000-0000-000000000000")!,
+        name: "",
+        createdAt: Date(timeIntervalSinceReferenceDate: 1234567890)
+      )
+    ]
+  )
+}
+```
+
+Here we controlled the `date` dependency to always return the same date, and we controlled the
+`uuid` dependency to return an auto-incrementing UUID every time it is invoked, and we even 
+controlled the `clock` dependency using an [`ImmediateClock`][immediate-clock-docs] to squash all
+of time into a single instant. If we did not control these dependencies this test would be very 
+difficult to write since there is no way to accurately predict what will be returned by `Date()` 
+and `UUID()`, and we'd have to wait for real world time to pass, making the test slow.
+
+But, controllable dependencies aren't only useful for tests. They can also be used in Xcode
+previews. Suppose the feature above makes use of a clock to sleep for an amount of time before
+something happens in the view. If you don't want to literally wait for time to pass in order to see
+how the view changes, you can override the clock dependency to be an "immediate" clock using the
+[`withDependencies`][withdependencies-docs] helper:
+
+```swift
+struct Feature_Previews: PreviewProvider {
+  static var previews: some View {
+    FeatureView(
+      model: withDependencies {
+        $0.clock = ImmediateClock()
+      } operation: {
+        FeatureModel()
+      }
+    )
+  }
+}
+```
+
+This will make it so that the preview uses an immediate clock when run, but when running in a
+simulator or on device it will still use a live `ContinuousClock`. This makes it possible to
+override dependencies just for previews without affecting how your app will run in production.
+
+That is the basics to getting started with using the library, but there is still a lot more you
+can do. You can learn more in depth about the library by exploring the [documentation][docs]
+and articles:
+
+#### Getting started
 
 * **[Quick start][quick-start-article]**:
   Learn the basics of getting started with the library before diving deep into all of its features.
```

Comment:
Wonder if we should remove this or move the link up (with "(Excerpted below.)" or something) to avoid redundancy?